MerlinAU.sh

```diff
@@ -5293,25 +5292,44 @@ _EntwareServicesHandler_()
 }
 
 ##------------------------------------------##
-## Modified by ExtremeFiretop [2024-Oct-13] ##
+## Modified by ExtremeFiretop [2024-Nov-15] ##
 ##------------------------------------------##
 _GetOfflineFirmwareVersion_()
 {
     local zip_file="$1"
     local extract_version_regex='[0-9]+_[0-9]+\.[0-9]+_[0-9a-zA-Z]+'
     local validate_version_regex='[0-9]+\.[0-9]+\.[0-9]+\.[0-9]+(_[0-9a-zA-Z]+)?'
     local fwVersionFormat
+    local firmware_version
+    local formatted_version
 
     # Extract the version number using regex #
     firmware_version="$(echo "$zip_file" | grep -oE "$extract_version_regex")"
 
-    if [ -n "$firmware_version" ]
-    then
-        # Replace underscores with dots and insert .0 before the suffix #
-        formatted_version="$(echo "$firmware_version" | sed -E 's/^([0-9]+)_([0-9]+\.[0-9]+)_([0-9a-zA-Z]+)/\1.\2.0_\3/')"
+    if [ -n "$firmware_version" ]; then
+        if echo "$firmware_version" | grep -qE '^([0-9]+)_([0-9]+)\.([0-9]+)_([0-9]+)$'; then
+            # Numeric patch version
+            formatted_version="$(echo "$firmware_version" | sed -E 's/^([0-9]+)_([0-9]+)\.([0-9]+)_([0-9]+)/\1.\2.\3.\4/')"
+        elif echo "$firmware_version" | grep -qE '^([0-9]+)_([0-9]+)\.([0-9]+)_([0-9a-zA-Z]+)$'; then
+            # Alphanumeric suffix
+            formatted_version="$(echo "$firmware_version" | sed -E 's/^([0-9]+)_([0-9]+)\.([0-9]+)_([0-9a-zA-Z]+)/\1.\2.\3.0_\4/')"
+        else
+            printf "\nFailed to parse firmware version from the ZIP file name.\n"
+            firmware_version=""
+        fi
         printf "\nIdentified firmware version: ${GRNct}$formatted_version${NOct}\n"
         printf "\n---------------------------------------------------\n"
-    else
+
+        # Ask the user to confirm the detected firmware version
+        if _WaitForYESorNO_ "\nIs this firmware version correct?"; then
+            printf "\n---------------------------------------------------\n"
+        else
+            # Set firmware_version to empty to trigger manual entry
+            firmware_version=""
+        fi
+    fi
```

Comment:
Confirm if the auto-detection version is correct, otherwise allow the user to enter manually.